### ISSUE TITLE
CKPT server core dump because of a race in replica intimation

### ISSUE THREAD

**Issue by karthick18** [OpenClovis/SAFplus-Availability-Scalability-Platform#42]
CKPT server has a rare race in ckptReplicaInfoUpdate while trying to fetch the checkpoint information from active replica on a checkpoint open or rebalancing during node arrival.
The race stems from the fact that if the replcia intimation gets delayed for the same checkpoint and retried ( > 10 seconds response times), then there is a possibility of 2 threads doing the replica update for the same checkpoint with 1 thread deleting/updating the last instance of the checkpoint before updating the replica with the new info. in update.
This would result in a core-dump as a result of memory corruption in another thread trying to update the replica information pertaining to the same checkpoint freed from another thread.
Here is a backtrace obtained from a customer running on 5.1 code base where the delay seems to have been introduced by his tests while debugging their apps.

Thread 8 is freeing the pCkpt while Thread 1 that took the fault is accessing it.
Hi Karthick,

We encountered a crash on a payload blade while debugging it (essentially restarting one of the components – ospf).  This is not a priority issue since our software was being debugged at the time (and has happened only once) but I thought I would pass along the stack traces.  Could this by chance be a known issue?   I don’t have logs for this one, only the core file.

Regards,

Jonathan

Core was generated by `asp_ckpt'.
Program terminated with signal 11, Segmentation fault.
#0 0x000000000044b9d5 in _ckptPresenceListPeerUpdate (pCkpt=0x1cf2c58, ckptHdl=1970367786647554, peerAddr=5) at clCkptPeer.c:1663

1663 clCkptPeer.c: No such file or directory.
in clCkptPeer.c
(gdb)
bt
#0 0x000000000044b9d5 in _ckptPresenceListPeerUpdate (pCkpt=0x1cf2c58, ckptHdl=1970367786647554, peerAddr=5) at clCkptPeer.c:1663
#1 0x000000000044f373 in _ckptReplicaInfoUpdate (ckptHdl=1970367786647554, pCkptInfo=0x403393a0) at clCkptPeer.c:2380
#2 0x000000000041ed79 in clCkptReplicaNotify_4_0_0 (pVersion=0x40339470, ckptActHdl=1970367786647554, activeAddr=7) at clCkptSaf.c:3845
#3 0x0000000000462b10 in clCkptReplicaNotifyServer_4_0_0 (eoData=0x0, inMsgHdl=0x1cdcb40, outMsgHdl=0x1b76b10) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/server/ckptEockptServerMasterActiveServer.c:999
#4 0x00007f20068342ad in clRmdInvoke (func=0x4629e4 <clCkptReplicaNotifyServer_4_0_0>, eoArg=0x0, inMsgHdl=0x1cdcb40, outMsgHdl=0x1b76b10) at clRmdHandle.c:143
#5 0x00007f2006a5574e in clEoWalkWithVersion (pThis=0x1cb49d8, func=25, version=0x40339a00, pFuncCallout=0x7f2006834104 <clRmdInvoke>, inMsgHdl=0x1cdcb40, outMsgHdl=0x1b76b10) at eo.c:2372
#6 0x00007f20068371c1 in rmdHandleAsyncRequest (pThis=0x1cb49d8, pReq=0x40339c60, srcAddr=0x40339b2c, priority=0 '\000', protoType=18 '\022', inMsgHdl=0x1cdcb40) at clRmdRecv.c:428
#7 0x00007f2006836768 in clRmdReceiveAsyncRequest (pThis=0x1cb49d8, rmdRecvMsg=0x1cdcb40, priority=0 '\000', protoType=18 '\022', length=56, srcAddr=...) at clRmdRecv.c:248
#8 0x00007f2006a59ae2 in clEoJobHandler (pJob=0x1cd6cc8) at eo.c:3796
#9 0x00007f200798f9df in clTaskPoolEntry (pArg=0x1b74e28) at clTaskPool.c:280
#10 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1b74e68) at posix/clCommonCos.c:956
#11 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#12 0x00007f20035f664d in clone () from /lib/libc.so.6
#13 0x0000000000000000 in ?? ()

thread apply all bt

Thread 15 (Thread 0x42022950 (LWP 3152)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x42020b78, mutexId=0x1cbbc08, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x42020b78, mutexId=0x1cbbc08, timer=...) at osal.c:678
#3 0x00007f200683ec44 in clRmdSyncSendAndReplyReceive (pThis=0x1cb49d8, pRmdObject=0x1cbbb78, hdrLen=32, outMsgHdl=0x1cd91a0, pOptions=0x42020ff0, nwMsgId=24, inMsg=0x1cd8180, destAddr=..., flags=26) at clRmd.c:1231
#4 0x00007f200683ce4f in clRmdWithMessage (remoteObjAddr=..., version=0x42021340, funcId=38, inMsgHdl=0x1cd8180, outMsgHdl=0x1cd91a0, flags=26, pOptions=0x7f1ffc020d58, pAsyncOptions=0x0) at clRmd.c:737
#5 0x00007f200683bafd in clRmdWithMsgVer (remoteObjAddr=..., version=0x42021340, funcId=38, inMsgHdl=0x1cd8180, outMsgHdl=0x1cd91a0, flags=26, pOptions=0x7f1ffc020d58, pAsyncOptions=0x0) at clRmd.c:446
#6 0x000000000046af57 in clCkptRemSvrCkptInfoGetClientSync_5_0_0 (handle=1407387768455170, pVersion=0x42021470, ckptActHdl=1970367786647556, peerAddr=5, pCkptInfo=0x420213a0) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/client/ckptEockptServerActivePeerClient.c:790
#7 0x000000000041ec3c in clCkptReplicaNotify_4_0_0 (pVersion=0x42021470, ckptActHdl=1970367786647556, activeAddr=7) at clCkptSaf.c:3808
#8 0x0000000000462b10 in clCkptReplicaNotifyServer_4_0_0 (eoData=0x0, inMsgHdl=0x7f1ffc030c00, outMsgHdl=0x7f1ffc0359f0) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/server/ckptEockptServerMasterActiveServer.c:999
#9 0x00007f20068342ad in clRmdInvoke (func=0x4629e4 <clCkptReplicaNotifyServer_4_0_0>, eoArg=0x0, inMsgHdl=0x7f1ffc030c00, outMsgHdl=0x7f1ffc0359f0) at clRmdHandle.c:143
#10 0x00007f2006a5574e in clEoWalkWithVersion (pThis=0x1cb49d8, func=25, version=0x42021a00, pFuncCallout=0x7f2006834104 <clRmdInvoke>, inMsgHdl=0x7f1ffc030c00, outMsgHdl=0x7f1ffc0359f0) at eo.c:2372
#11 0x00007f20068371c1 in rmdHandleAsyncRequest (pThis=0x1cb49d8, pReq=0x42021c60, srcAddr=0x42021b2c, priority=0 '\000', protoType=18 '\022', inMsgHdl=0x7f1ffc030c00) at clRmdRecv.c:428
#12 0x00007f2006836768 in clRmdReceiveAsyncRequest (pThis=0x1cb49d8, rmdRecvMsg=0x7f1ffc030c00, priority=0 '\000', protoType=18 '\022', length=56, srcAddr=...) at clRmdRecv.c:248
#13 0x00007f2006a59ae2 in clEoJobHandler (pJob=0x7f1ffc013608) at eo.c:3796
#14 0x00007f200798f9df in clTaskPoolEntry (pArg=0x1cd80f8) at clTaskPool.c:280
#15 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x7f1ffc02e368) at posix/clCommonCos.c:956
#16 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#17 0x00007f20035f664d in clone () from /lib/libc.so.6
#18 0x0000000000000000 in ?? ()

Thread 14 (Thread 0x41072950 (LWP 3148)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x1cd35e8, mutexId=0x1cd35b8, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x1cd35e8, mutexId=0x1cd35b8, timer=...) at osal.c:678
#3 0x00007f200798f7b1 in clTaskPoolEntry (pArg=0x1b76228) at clTaskPool.c:193
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6ce8) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 13 (Thread 0x40856950 (LWP 3045)):
#0 0x00007f20035edbd6 in poll () from /lib/libc.so.6
#1 0x00007f20065f7086 in __clIocReceive (commPortHdl=30128744, pRecvOption=0x40855ed0, buffer=0x40845d50 "\001\026", bufSize=65536, message=0x1cd6d70, pRecvParam=0x40855ec0, sync=0) at clTipcUserApi.c:2572
#2 0x00007f20065f80c2 in clIocReceiveAsync (commPortHdl=30128744, pRecvOption=0x40855ed0, message=0x1cd6d70, pRecvParam=0x40855ec0) at clTipcUserApi.c:2833
#3 0x00007f2006a591cf in clEoIocRecvQueueProcess (pThis=0x1cb49d8) at eo.c:3644
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6848) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 12 (Thread 0x7f200860b6f0 (LWP 3034)):
#0 0x00007f2003d97d29 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75427 in cosPosixCondWait (conditionId=0x1cb4a90, mutexId=0x1cb4a60, timer=...) at posix/clCommonCos.c:1554
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x1cb4a90, mutexId=0x1cb4a60, timer=...) at osal.c:678
#3 0x00007f2006a5cba2 in clEoMain (argc=1, argv=0x7fffb9c6b1f8) at clEo.c:858
#4 0x00007f2006a5cf5b in clEoInitialize (argc=1, argv=0x7fffb9c6b1f8) at clEoMain.c:175
#5 0x00007f20050835e6 in main (argc=1, argv=0x7fffb9c6b1f8) at clMain.c:48
#6 0x00007f20035451a6 in __libc_start_main () from /lib/libc.so.6
#7 0x000000000040f179 in _start ()

Thread 11 (Thread 0x41e5e950 (LWP 3150)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x1cd35e8, mutexId=0x1cd35b8, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x1cd35e8, mutexId=0x1cd35b8, timer=...) at osal.c:678
#3 0x00007f200798f7b1 in clTaskPoolEntry (pArg=0x7f1ffc02e2b8) at clTaskPool.c:193
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x7f1ffc030278) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 10 (Thread 0x4010a950 (LWP 3042)):
#0 0x00007f2003d97d29 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75427 in cosPosixCondWait (conditionId=0x7f2006c67e88, mutexId=0x7f2006c67eb8, timer=...) at posix/clCommonCos.c:1554
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x7f2006c67e88, mutexId=0x7f2006c67eb8, timer=...) at osal.c:678
#3 0x00007f2006a4eb42 in eoStaticQueueProcess (pThis=0x1cb49d8) at eo.c:700
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6848) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 9 (Thread 0x402a1950 (LWP 3050)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x1cd4fe8, mutexId=0x1cd4fb8, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x1cd4fe8, mutexId=0x1cd4fb8, timer=...) at osal.c:678
#3 0x00007f200798f7b1 in clTaskPoolEntry (pArg=0x1b76248) at clTaskPool.c:193
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1b758d8) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 8 (Thread 0x405d7950 (LWP 3091)):
#0 0x00007f2003d954f8 in pthread_mutex_lock () from /lib/libpthread.so.0
#1 0x00007f2005d68149 in __cosMutexLock (mutexId=0x7f2007bcf570, verbose=1) at posix/clLinux.c:328
#2 0x00007f2005d6819c in cosMutexLock (mutexId=0x7f2007bcf570) at posix/clLinux.c:342
#3 0x00007f2005d60496 in clOsalMutexLock (mutexId=0x7f2007bcf570) at osal.c:537
#4 0x00007f200796a55d in clEoMemWaterMarksUpdate (memDir=CL_MEM_FREE) at clMemStats.c:295
#5 0x00007f2007966209 in clHeapFree (pAddress=0x1cfbf20) at clHeap.c:1650
#6 0x00007f20061b4ac4 in cclLinkedListContainerDestroy (containerHandle=0x1cfbf28) at clovisLinkedList.c:1028
#7 0x00007f20061a96ab in clCntDelete (containerHandle=0x1cfbf28) at clovisContainer.c:674
#8 0x00007f200619baa4 in cclHashTableContainerDestroy (containerHandle=0x1ce2ba8) at clovisHashTable.c:329
#9 0x00007f20061a96ab in clCntDelete (containerHandle=0x1ce2ba8) at clovisContainer.c:674
#10 0x0000000000423a07 in _ckptDplaneInfoFree (pDpInfo=0x1ce2b58) at clCkptUtils.c:986
#11 0x00000000004236aa in ckptEntryFree (pCkpt=0x1cf2c58) at clCkptUtils.c:841
#12 0x00000000004219bf in ckptSvrHdlDeleteCallback (userKey=0xfa129c54, userData=0x1cf2ab8) at clCkptUtils.c:253
#13 0x00007f20061ad7e5 in cclLinkedListContainerNodeDelete (containerHandle=0x7f1ffc013728, nodeHandle=0x1cf2bc8) at clovisLinkedList.c:591
#14 0x00007f20061a46bc in clCntNodeDelete (containerHandle=0x7f1ffc013728, nodeHandle=0x1cf2bc8) at clovisContainer.c:253
#15 0x00007f20061a97ae in clCntNonUniqueKeyDelete (container=0x7f1ffc013728, key=0xfa129c54, givenData=0x1cef128, cmp=0x427441 <ckptHdlNonUniqueKeyCompare>) at clovisContainer.c:702
#16 0x000000000044f413 in _ckptReplicaInfoUpdate (ckptHdl=1970367786647554, pCkptInfo=0x405d63a0) at clCkptPeer.c:2408
#17 0x000000000041ed79 in clCkptReplicaNotify_4_0_0 (pVersion=0x405d6470, ckptActHdl=1970367786647554, activeAddr=7) at clCkptSaf.c:3845
#18 0x0000000000462b10 in clCkptReplicaNotifyServer_4_0_0 (eoData=0x0, inMsgHdl=0x1cd7850, outMsgHdl=0x1cd8990) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/server/ckptEockptServerMasterActiveServer.c:999
#19 0x00007f20068342ad in clRmdInvoke (func=0x4629e4 <clCkptReplicaNotifyServer_4_0_0>, eoArg=0x0, inMsgHdl=0x1cd7850, outMsgHdl=0x1cd8990) at clRmdHandle.c:143
#20 0x00007f2006a5574e in clEoWalkWithVersion (pThis=0x1cb49d8, func=25, version=0x405d6a00, pFuncCallout=0x7f2006834104 <clRmdInvoke>, inMsgHdl=0x1cd7850, outMsgHdl=0x1cd8990) at eo.c:2372
#21 0x00007f20068371c1 in rmdHandleAsyncRequest (pThis=0x1cb49d8, pReq=0x405d6c60, srcAddr=0x405d6b2c, priority=0 '\000', protoType=18 '\022', inMsgHdl=0x1cd7850) at clRmdRecv.c:428
#22 0x00007f2006836768 in clRmdReceiveAsyncRequest (pThis=0x1cb49d8, rmdRecvMsg=0x1cd7850, priority=0 '\000', protoType=18 '\022', length=56, srcAddr=...) at clRmdRecv.c:248
#23 0x00007f2006a59ae2 in clEoJobHandler (pJob=0x1cd6c68) at eo.c:3796
#24 0x00007f200798f9df in clTaskPoolEntry (pArg=0x1cd6cc8) at clTaskPool.c:280
#25 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6ce8) at posix/clCommonCos.c:956
#26 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#27 0x00007f20035f664d in clone () from /lib/libc.so.6
#28 0x0000000000000000 in ?? ()

Thread 7 (Thread 0x40e83950 (LWP 3149)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x40e81b78, mutexId=0x1cbbc08, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x40e81b78, mutexId=0x1cbbc08, timer=...) at osal.c:678
#3 0x00007f200683ec44 in clRmdSyncSendAndReplyReceive (pThis=0x1cb49d8, pRmdObject=0x1cbbb78, hdrLen=32, outMsgHdl=0x1cde370, pOptions=0x40e81ff0, nwMsgId=20, inMsg=0x1cddb60, destAddr=..., flags=26) at clRmd.c:1231
#4 0x00007f200683ce4f in clRmdWithMessage (remoteObjAddr=..., version=0x40e82340, funcId=38, inMsgHdl=0x1cddb60, outMsgHdl=0x1cde370, flags=26, pOptions=0x7f1ffc020d58, pAsyncOptions=0x0) at clRmd.c:737
#5 0x00007f200683bafd in clRmdWithMsgVer (remoteObjAddr=..., version=0x40e82340, funcId=38, inMsgHdl=0x1cddb60, outMsgHdl=0x1cde370, flags=26, pOptions=0x7f1ffc020d58, pAsyncOptions=0x0) at clRmd.c:446
#6 0x000000000046af57 in clCkptRemSvrCkptInfoGetClientSync_5_0_0 (handle=1407387768455170, pVersion=0x40e82470, ckptActHdl=1970367786647556, peerAddr=5, pCkptInfo=0x40e823a0) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/client/ckptEockptServerActivePeerClient.c:790
#7 0x000000000041ec3c in clCkptReplicaNotify_4_0_0 (pVersion=0x40e82470, ckptActHdl=1970367786647556, activeAddr=7) at clCkptSaf.c:3808
#8 0x0000000000462b10 in clCkptReplicaNotifyServer_4_0_0 (eoData=0x0, inMsgHdl=0x1cda1c0, outMsgHdl=0x1cdc330) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/server/ckptEockptServerMasterActiveServer.c:999
#9 0x00007f20068342ad in clRmdInvoke (func=0x4629e4 <clCkptReplicaNotifyServer_4_0_0>, eoArg=0x0, inMsgHdl=0x1cda1c0, outMsgHdl=0x1cdc330) at clRmdHandle.c:143
#10 0x00007f2006a5574e in clEoWalkWithVersion (pThis=0x1cb49d8, func=25, version=0x40e82a00, pFuncCallout=0x7f2006834104 <clRmdInvoke>, inMsgHdl=0x1cda1c0, outMsgHdl=0x1cdc330) at eo.c:2372
#11 0x00007f20068371c1 in rmdHandleAsyncRequest (pThis=0x1cb49d8, pReq=0x40e82c60, srcAddr=0x40e82b2c, priority=0 '\000', protoType=18 '\022', inMsgHdl=0x1cda1c0) at clRmdRecv.c:428
#12 0x00007f2006836768 in clRmdReceiveAsyncRequest (pThis=0x1cb49d8, rmdRecvMsg=0x1cda1c0, priority=0 '\000', protoType=18 '\022', length=56, srcAddr=...) at clRmdRecv.c:248
#13 0x00007f2006a59ae2 in clEoJobHandler (pJob=0x1b758d8) at eo.c:3796
#14 0x00007f200798f9df in clTaskPoolEntry (pArg=0x1cd8158) at clTaskPool.c:280
#15 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6ce8) at posix/clCommonCos.c:956
#16 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#17 0x00007f20035f664d in clone () from /lib/libc.so.6
#18 0x0000000000000000 in ?? ()

Thread 6 (Thread 0x41c27950 (LWP 3036)):
#0 0x00007f2003d9b0e1 in nanosleep () from /lib/libpthread.so.0
#1 0x00007f2005d6e2db in cosPosixTaskDelay (timer=...) at posix/clCommonCos.c:810
#2 0x00007f2005d5d3fc in clOsalTaskDelay (timer=...) at osal.c:275
#3 0x00007f2005f93258 in clTimerTask (pArg=0x0) at clTimerTree.c:1298
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1b79048) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 5 (Thread 0x4079e950 (LWP 3046)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x1cd57b8, mutexId=0x1cd5788, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x1cd57b8, mutexId=0x1cd5788, timer=...) at osal.c:678
#3 0x00007f200798f7b1 in clTaskPoolEntry (pArg=0x1cd6b18) at clTaskPool.c:193
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6848) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 4 (Thread 0x408db950 (LWP 3038)):
#0 0x00007f2003d97d29 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75427 in cosPosixCondWait (conditionId=0x7f2007bdd720, mutexId=0x7f2007bdd760, timer=...) at posix/clCommonCos.c:1554
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x7f2007bdd720, mutexId=0x7f2007bdd760, timer=...) at osal.c:678
#3 0x00007f2007988d72 in clLogDeferredHandler (pData=0x0) at clLogUtils.c:453
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1b78238) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 3 (Thread 0x40988950 (LWP 3092)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x1cd35e8, mutexId=0x1cd35b8, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x1cd35e8, mutexId=0x1cd35b8, timer=...) at osal.c:678
#3 0x00007f200798f7b1 in clTaskPoolEntry (pArg=0x1cd6cc8) at clTaskPool.c:193
#4 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6ce8) at posix/clCommonCos.c:956
#5 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#6 0x00007f20035f664d in clone () from /lib/libc.so.6
#7 0x0000000000000000 in ?? ()

Thread 2 (Thread 0x40887950 (LWP 3147)):
#0 0x00007f2003d97fad in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/libpthread.so.0
#1 0x00007f2005d75811 in cosPosixCondWait (conditionId=0x40885b78, mutexId=0x1cbbc08, timer=...) at posix/clCommonCos.c:1572
#2 0x00007f2005d62449 in clOsalCondWait (conditionId=0x40885b78, mutexId=0x1cbbc08, timer=...) at osal.c:678
#3 0x00007f200683ec44 in clRmdSyncSendAndReplyReceive (pThis=0x1cb49d8, pRmdObject=0x1cbbb78, hdrLen=32, outMsgHdl=0x1cdb9f0, pOptions=0x40885ff0, nwMsgId=19, inMsg=0x1cdb1e0, destAddr=..., flags=26) at clRmd.c:1231
#4 0x00007f200683ce4f in clRmdWithMessage (remoteObjAddr=..., version=0x40886340, funcId=38, inMsgHdl=0x1cdb1e0, outMsgHdl=0x1cdb9f0, flags=26, pOptions=0x7f1ffc020d58, pAsyncOptions=0x0) at clRmd.c:737
#5 0x00007f200683bafd in clRmdWithMsgVer (remoteObjAddr=..., version=0x40886340, funcId=38, inMsgHdl=0x1cdb1e0, outMsgHdl=0x1cdb9f0, flags=26, pOptions=0x7f1ffc020d58, pAsyncOptions=0x0) at clRmd.c:446
#6 0x000000000046af57 in clCkptRemSvrCkptInfoGetClientSync_5_0_0 (handle=1407387768455170, pVersion=0x40886470, ckptActHdl=1970367786647555, peerAddr=5, pCkptInfo=0x408863a0) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/client/ckptEockptServerActivePeerClient.c:790
#7 0x000000000041ec3c in clCkptReplicaNotify_4_0_0 (pVersion=0x40886470, ckptActHdl=1970367786647555, activeAddr=7) at clCkptSaf.c:3808
#8 0x0000000000462b10 in clCkptReplicaNotifyServer_4_0_0 (eoData=0x0, inMsgHdl=0x1b76300, outMsgHdl=0x1cda9d0) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/server/ckptEockptServerMasterActiveServer.c:999
#9 0x00007f20068342ad in clRmdInvoke (func=0x4629e4 <clCkptReplicaNotifyServer_4_0_0>, eoArg=0x0, inMsgHdl=0x1b76300, outMsgHdl=0x1cda9d0) at clRmdHandle.c:143
#10 0x00007f2006a5574e in clEoWalkWithVersion (pThis=0x1cb49d8, func=25, version=0x40886a00, pFuncCallout=0x7f2006834104 <clRmdInvoke>, inMsgHdl=0x1b76300, outMsgHdl=0x1cda9d0) at eo.c:2372
#11 0x00007f20068371c1 in rmdHandleAsyncRequest (pThis=0x1cb49d8, pReq=0x40886c60, srcAddr=0x40886b2c, priority=0 '\000', protoType=18 '\022', inMsgHdl=0x1b76300) at clRmdRecv.c:428
#12 0x00007f2006836768 in clRmdReceiveAsyncRequest (pThis=0x1cb49d8, rmdRecvMsg=0x1b76300, priority=0 '\000', protoType=18 '\022', length=56, srcAddr=...) at clRmdRecv.c:248
#13 0x00007f2006a59ae2 in clEoJobHandler (pJob=0x1b758a8) at eo.c:3796
#14 0x00007f200798f9df in clTaskPoolEntry (pArg=0x1cd6cc8) at clTaskPool.c:280
#15 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1cd6ce8) at posix/clCommonCos.c:956
#16 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#17 0x00007f20035f664d in clone () from /lib/libc.so.6
#18 0x0000000000000000 in ?? ()

Thread 1 (Thread 0x4033a950 (LWP 3151)):
#0 0x000000000044b9d5 in _ckptPresenceListPeerUpdate (pCkpt=0x1cf2c58, ckptHdl=1970367786647554, peerAddr=5) at clCkptPeer.c:1663
#1 0x000000000044f373 in _ckptReplicaInfoUpdate (ckptHdl=1970367786647554, pCkptInfo=0x403393a0) at clCkptPeer.c:2380
#2 0x000000000041ed79 in clCkptReplicaNotify_4_0_0 (pVersion=0x40339470, ckptActHdl=1970367786647554, activeAddr=7) at clCkptSaf.c:3845
#3 0x0000000000462b10 in clCkptReplicaNotifyServer_4_0_0 (eoData=0x0, inMsgHdl=0x1cdcb40, outMsgHdl=0x1b76b10) at /anroot/projects/openclovis/openclovis/affirmed/sdk-5.1/src/ASP/components/ckpt/idl/ckptEo/server/ckptEockptServerMasterActiveServer.c:999
#4 0x00007f20068342ad in clRmdInvoke (func=0x4629e4 <clCkptReplicaNotifyServer_4_0_0>, eoArg=0x0, inMsgHdl=0x1cdcb40, outMsgHdl=0x1b76b10) at clRmdHandle.c:143
#5 0x00007f2006a5574e in clEoWalkWithVersion (pThis=0x1cb49d8, func=25, version=0x40339a00, pFuncCallout=0x7f2006834104 <clRmdInvoke>, inMsgHdl=0x1cdcb40, outMsgHdl=0x1b76b10) at eo.c:2372
#6 0x00007f20068371c1 in rmdHandleAsyncRequest (pThis=0x1cb49d8, pReq=0x40339c60, srcAddr=0x40339b2c, priority=0 '\000', protoType=18 '\022', inMsgHdl=0x1cdcb40) at clRmdRecv.c:428
#7 0x00007f2006836768 in clRmdReceiveAsyncRequest (pThis=0x1cb49d8, rmdRecvMsg=0x1cdcb40, priority=0 '\000', protoType=18 '\022', length=56, srcAddr=...) at clRmdRecv.c:248
#8 0x00007f2006a59ae2 in clEoJobHandler (pJob=0x1cd6cc8) at eo.c:3796
#9 0x00007f200798f9df in clTaskPoolEntry (pArg=0x1b74e28) at clTaskPool.c:280
#10 0x00007f2005d6f52b in cosPosixTaskWrapper (pArgument=0x1b74e68) at posix/clCommonCos.c:956
#11 0x00007f2003d93fc7 in start_thread () from /lib/libpthread.so.0
#12 0x00007f20035f664d in clone () from /lib/libc.so.6
#13 0x0000000000000000 in ?? ()

(gdb)
root@an3000-mcm-slot7cpu1:/var/corefiles/slot5cpu1#
